components/masking_functions/src/masking_functions/dictionary_flusher_thread.cpp

```diff
@@ -359,6 +366,21 @@ void dictionary_flusher_thread::do_periodic_reload() {
 
       failure_message.reset();
       try {
+        // In rare cases when this thread is in the middle of execution of
```

Comment:
As discussed on Slack, I am not quite convinced that after your changes above we need this resetting because of the reason explained in this comment.

OTOH it should not harm. So perhaps it is better to replace this comment with more generic one. Something like:

// Play safe. Avoid assertion failures in case when we query dictionaries table after previous attempt has been aborted.

What do you think?

IMO we also should extend our test coverage to test situation when dictionaries table is dropped while flusher thread is active.